facade/nodeFacade.go

```diff
@@ -391,49 +398,144 @@ func (nf *nodeFacade) GetNumCheckpointsFromAccountState() uint32 {
 }
 
 // GetProof returns the Merkle proof for the given address and root hash
-func (nf *nodeFacade) GetProof(rootHash string, address string) ([][]byte, error) {
-	rootHashBytes, err := hex.DecodeString(rootHash)
+func (nf *nodeFacade) GetProof(rootHash string, address string) (*shared.GetProofResponse, error) {
+	proof, value, err := nf.getProof(rootHash, address)
 	if err != nil {
 		return nil, err
 	}
 
+	return &shared.GetProofResponse{
+		Proof:    proof,
+		Value:    value,
+		RootHash: rootHash,
+	}, nil
+}
+
+func (nf *nodeFacade) getProof(rootHash string, key string) ([][]byte, []byte, error) {
+	rootHashBytes, err := hex.DecodeString(rootHash)
+	if err != nil {
+		return nil, nil, err
+	}
+
 	trie, err := nf.accountsState.GetTrie(rootHashBytes)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	addressBytes, err := nf.DecodeAddressPubkey(address)
+	keyBytes, err := nf.getKeyBytes(key)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return trie.GetProof(addressBytes)
+	computedProof, err := trie.GetProof(keyBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	value, err := trie.Get(keyBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return computedProof, value, nil
 }
 
-// GetProofCurrentRootHash returns the Merkle proof for the given address and current root hash
-func (nf *nodeFacade) GetProofCurrentRootHash(address string) ([][]byte, []byte, error) {
-	currentBlockHeader := nf.blockchain.GetCurrentBlockHeader()
-	if check.IfNil(currentBlockHeader) {
-		return nil, nil, ErrNilBlockHeader
+func (nf *nodeFacade) getKeyBytes(key string) ([]byte, error) {
+	addressBytes, err := nf.DecodeAddressPubkey(key)
+	if err == nil {
+		return addressBytes, nil
 	}
 
-	rootHash := currentBlockHeader.GetRootHash()
-	trie, err := nf.accountsState.GetTrie(rootHash)
+	return hex.DecodeString(key)
+}
+
+func (nf *nodeFacade) getDataTrieRootHash(accountBytes []byte) []byte {
+	account := &state.UserAccountData{}
+	err := nf.marshalizer.Unmarshal(account, accountBytes)
+	if err != nil {
+		return nil
+	}
+
+	return account.RootHash
+}
+
+// GetProofDataTrie returns the Merkle Proof for the given address, and another Merkle Proof
+// for the given key, if it is present in the dataTrie
```

Comment:
"if it is present in the dataTrie" -> "if it exists in the dataTrie"

Reply:
Done.